src/parser.cc

```diff
@@ -754,16 +755,16 @@ std::string getModelFilePath(const std::string &_modelDirPath)
     }
   }
 
-  TiXmlDocument configFileDoc;
-  if (!configFileDoc.LoadFile(configFilePath))
+  tinyxml2::XMLDocument configFileDoc;
+  if (!configFileDoc.LoadFile(configFilePath.c_str()))
```

Comment:
This should actually be the opposite logic. Consider comparing the return value against tinyxml2::XML_SUCCESS. Changing this to
```suggestion
  if (tinyxml2::XML_SUCCESS != configFileDoc.LoadFile(configFilePath.c_str()))
```
fixes all the test failures.